src/server/rest/v1/service/ChargingStationService.ts

```diff
@@ -1152,6 +1168,51 @@ export default class ChargingStationService {
     next();
   }
 
+  // For Testing
+  public static async authorizeOICPUser(tenant: Tenant, user: User, chargingStation: ChargingStation, connectorId: number, tagID: string, action: ServerAction): Promise<boolean> {
+    const oicpClient = await OICPClientFactory.getAvailableOicpClient(tenant, OICPRole.CPO) as CpoOICPClient;
+    if (!oicpClient) {
+      throw new BackendError({
+        user: user,
+        action: action,
+        module: MODULE_NAME,
+        method: 'isTagIDAuthorizedOnChargingStation',
+        message: 'OICP component requires at least one CPO endpoint to start a Session'
+      });
+    }
+    // Call Hubject
+    const oicpSession = {} as OICPSession;
+    oicpSession.identification = OICPUtils.convertTagID2OICPIdentification(tagID);
+    const evse = {} as OICPEvseDataRecord;
+    const connector = Utils.getConnectorFromID(chargingStation, connectorId);
+    evse.EvseID = OICPUtils.buildEvseID(oicpClient.getLocalCountryCode(ServerAction.CHARGING_STATION_REMOTE_START_TRANSACTION), oicpClient.getLocalPartyID(ServerAction.CHARGING_STATION_REMOTE_START_TRANSACTION), chargingStation, connector);
+    oicpSession.evse = evse;
+    const response = await oicpClient.authorizeStart(oicpSession, user);
+    const existingAuthorization: RemoteAuthorization = chargingStation.remoteAuthorizations.find(
+      (authorization) => authorization.connectorId === connectorId);
+    if (existingAuthorization) {
+      existingAuthorization.timestamp = new Date();
+      // No authorization ID available
+      // Use sessionID instead
+      existingAuthorization.id = response.SessionID;
+      existingAuthorization.tagId = tagID;
+      existingAuthorization.oicpIdentification = oicpSession.identification;
+    } else {
+      chargingStation.remoteAuthorizations.push(
+        {
+          id: response.SessionID,
+          connectorId: connectorId,
+          timestamp: new Date(),
+          tagId: tagID,
+          oicpIdentification: oicpSession.identification
+        }
+      );
```

Comment:
Is that ChargingStation data structure add-on specific to OICP? or OICP also need a similar extension? 
I do think it should be added to the existing ChargingStation data structure and collection but should lie in a specific data structure and collection like 'roamingauthorizations' that will permit to keep their history and be easily linked to others records like charging stations or transactions.

Reply:
The oicpIdentification oobject is specific for OICP. It contains information about the authorization method RFID card, Plug&Charge, Remote etc. including TagId or EVCO ID and more

I need that information for the following steps.
I think it makes sense to save this kind of data also for OCPI